service/gcs/core/gcs/gcs.go

```diff
@@ -77,6 +77,7 @@ type containerCacheEntry struct {
 	ExitHooks          []func(oslayer.ProcessExitState)
 	MappedVirtualDisks map[uint8]prot.MappedVirtualDisk
 	NetworkAdapters    []prot.NetworkAdapter
+	container          runtime.Container
 }
```

Comment:
You can probably also remove the ID field here since a runtime.Container contains an ID.